Source/missiles.cpp

```diff
@@ -915,6 +876,100 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, bool resist, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (resist) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (mid > MAX_PLRS - 1)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)

```

Comment:
`Player` Considering that the function also accepts not being given a valid player id this function doesn't seam specific to players.

`AgainstMonster` is given by the last argument. This seams to be a good case for overloading where there could be a similar function that takes Player as the last argument.

Maybe a better name is `CalculateMissileChanceToHit()`, or even `Missile::CalculateChanceToHit()` so that it matches `Player::CalculateArmorPierce()` which it calls in turn.